qpc/insights/upload.py

```diff
@@ -0,0 +1,171 @@
+            sys.exit(1)
+
+    def _check_insights_version(self):
+        version_command = self.insights_command.version()
+        process = subprocess.Popen(version_command,
+                                   stderr=subprocess.PIPE,
+                                   stdout=subprocess.PIPE)
+        streamdata = format_subprocess_stderr(process)
+        code = process.returncode
+        version_check = check_insights_version(streamdata,
+                                               insights.CLIENT_VERSION,
+                                               insights.CORE_VERSION)
+        if not version_check['results'] or code is not 0:
+            if 'client' in version_check.keys():
+                print(_(messages.BAD_CLIENT_VERSION %
+                        (version_check['client'],
+                         insights.CLIENT_VERSION)))
+            if 'core' in version_check.keys():
+                print(_(messages.BAD_CORE_VERSION %
+                        (version_check['core'],
+                         insights.CORE_VERSION)))
+            if 'error' in version_check.keys():
+                print(_(messages.ERROR_INSIGHTS_VERSION %
+                        (version_check['error'])))
+            print(_(messages.CHECK_VERSION %
+                    (' '.join(version_command))))
+            sys.exit(1)
+
+    def _validate_args(self):
+        CliCommand._validate_args(self)
+        self.req_headers = {'Accept': 'application/json+gzip'}
+        if self.args.report_id is None:
+            response = request(parser=self.parser, method=GET,
+                               path='%s%s' % (scan.SCAN_JOB_URI,
+                                              self.args.scan_job_id),
+                               payload=None)
+            if response.status_code == codes.ok:  # pylint: disable=no-member
+                json_data = response.json()
+                self.report_id = json_data.get('report_id')
+                if self.report_id is None:
+                    print(_(messages.REPORT_NO_DEPLOYMENTS_REPORT_FOR_SJ %
+                            self.args.scan_job_id))
+                    sys.exit(1)
+            else:
+                print(_(messages.REPORT_SJ_DOES_NOT_EXIST %
+                        self.args.scan_job_id))
+                sys.exit(1)
+            self.report_source = 'scan job id(%s).' % self.args.scan_job_id
+        else:
+            self.report_id = self.args.report_id
+            self.report_source = 'id (%s).' % self.args.report_id
+        if self.args.test:
+            self.insights_command = InsightsCommands(test=True)
+        else:
+            self.insights_command = InsightsCommands()
+        self._check_insights_install()
+        self._check_insights_version()
+        print(_(messages.INSIGHTS_IS_VERIFIED))
+        try:
+            validate_write_file(self.tmp_tar_name, 'tmp_tar_name')
+        except ValueError as error:
+            print(error)
```

Comment:
let's try and construct a message to help guide the user here.

Reply:
Do you want me to do this for the deployments & details report too?

Reply:
`def validate_write_file(filename, param_name):
    """Write content to a file.

    :param filename: the filename to write
    :param param_name: the parameter name that provided
    the filename
    :raises: ValueError for validation errors
    """
    input_path = os.path.expanduser(os.path.expandvars(filename))
    if not input_path:
        raise ValueError(
            t(messages.REPORT_OUTPUT_CANNOT_BE_EMPTY % param_name))
    if os.path.isdir(input_path):
        raise ValueError(
            t(messages.REPORT_OUTPUT_IS_A_DIRECTORY %
              (param_name, input_path)))
    directory = os.path.dirname(input_path)
    if directory and not os.path.exists(directory):
        raise ValueError(
            t(messages.REPORT_DIRECTORY_DOES_NOT_EXIST % directory))`

I believe we overwrite the ValueError inside of `validate_write_file` to help guide the user.  We use the same format for details & deployments reports.

Reply:
Can you post the output?  I want to make sure the message is clean and not the printout of an object.  Let's not use the same format if it is not ideal.  I would lean towards changing that also.